Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -642,6 +654,42 @@ The following example specifies a new App Link using both the `http://` and `htt
 
 In this example, the `<intent-filter>` includes the flag `android:autoVerify="true"`, which makes it an App Link and causes the Android system to reach out to the declared `android:host` in an attempt to access the [Digital Asset Links file](https://developers.google.com/digital-asset-links/v1/getting-started "Digital Asset Link") in order to [verify the App Links](https://developer.android.com/training/app-links/verify-site-associations "Verify Android App Links").
 
+> It is an [Android best practice](https://developer.android.com/training/app-links/verify-site-associations#add-intent-filters) to include `autoVerify` to each `<intent-filter>`.  
+
+#### Check for Redirects
+
+The system doesn't verify any Android App Links for an app if the server sets a redirect such as "http://example.com to https://example.com" or "example.com to www.example.com".
+
+#### Check for Subdomains
+
+If an intent filter lists multiple hosts with different subdomains, there must be a valid assetlinks.json on each domain. For example, the following intent filter includes www.example.com and mobile.example.com as accepted intent URL hosts. So a valid assetlinks.json must be published at both https://www.example.com/.well-known/assetlinks.json and https://mobile.example.com/.well-known/assetlinks.json.
+
+Also note that if the hostname includes a wildcard (such as *.example.com), you should be able to find an assetlinks.json file at the root hostname (https://example.com/.well-known/assetlinks.json).
+
+```xml
+<application>
+  <activity android:name=”MainActivity”>
+    <intent-filter android:autoVerify="true">
+      <action android:name="android.intent.action.VIEW" />
+      <category android:name="android.intent.category.DEFAULT" />
+      <category android:name="android.intent.category.BROWSABLE" />
+      <data android:scheme="https" />
+      <data android:scheme="https" />
+      <data android:host="www.example.com" />
+      <data android:host="mobile.example.com" />
+    </intent-filter>
+  </activity>
+</application>
+```
+
+#### Check the Handler Method
+
+You should carefully analyze the logic of the deep link or App Link handler:
+
+- Check for **String Comparison Methods** being sued such as `startsWith`, `endsWith`, `contains` methods.
+- Check for [**Serialization Issues**](#testing-object-persistence-mstg-platform-8).
+- Check for **Data Transmission**
```

Comment:
That's pretty vague. What do you mean?

Reply:
"Data Transmission" continues below but I realize now that the flow is not right. I will rework this section a little bit.